### PR TITLE
Go screen, Speed & feed widgets, YetiPilot widgets, Spindle health check screen

### PR DESCRIPTION
**Pull request by archiepeterjarvis**
Manual changes applied in popup_yetipilot_settings.py

Tested on both console sizes.

### CODE REVIEW

File: src/asmcnc/skavaUI/screen_go.py [hunk 1]
```diff
@@ -539,425 +551,450 @@ def show_hide_yp_container(self, use_sc2):
...
+        self.sm.get_screen("spindle_shutdown").reason_for_pause = "job_pause"
+        self.sm.get_screen("spindle_shutdown").return_screen = "go"
+        self.sm.current = "spindle_shutdown"
 
     POLL_FOR_PAUSE_SCREENS = 0.5
 
     def raise_pause_screens_if_paused(self, dt=0, override=False):
-        # Ok so 'spindle_shutdown' & the above func needs renaming & refactoring,
-        # and the shutdown UI commands need pulling out of serial comms altogether, but that's for another day. 
-        # For now, this is enough:
```
Comment (LettyYeti) [inline src/asmcnc/skavaUI/screen_go.py:741]: .

File: src/asmcnc/skavaUI/screen_go.py [hunk 2]
```diff
@@ -539,425 +551,450 @@ def show_hide_yp_container(self, use_sc2):
...
-            self.sm.get_screen('spindle_shutdown').return_screen = "go"
-            self.sm.current = 'spindle_shutdown'
+            self.sm.get_screen(
+                "spindle_shutdown"
+            ).reason_for_pause = self.m.reason_for_machine_pause
+            self.sm.get_screen("spindle_shutdown").return_screen = "go"
+            self.sm.current = "spindle_shutdown"
             self.start_or_pause_button_image.source = "./asmcnc/skavaUI/img/go.png"
 
-    # Running
```
Comment (LettyYeti) [inline src/asmcnc/skavaUI/screen_go.py:760]: .

File: src/asmcnc/skavaUI/screen_go.py [hunk 3]
```diff
@@ -539,425 +551,450 @@ def show_hide_yp_container(self, use_sc2):
...
-
     def _start_running_job(self):
         self.database.send_job_start()
         self.jd.job_start_time = time.time()
-
         self.start_or_pause_button_image.source = "./asmcnc/skavaUI/img/pause.png"
-        # Hide back button
```
Comment (LettyYeti) [inline src/asmcnc/skavaUI/screen_go.py:767]: .

File: src/asmcnc/skavaUI/screen_go.py [hunk 4]
```diff
@@ -539,425 +551,450 @@ def show_hide_yp_container(self, use_sc2):
...
-        self.btn_back_img.source = './asmcnc/skavaUI/img/file_running.png'
+        self.btn_back_img.source = "./asmcnc/skavaUI/img/file_running.png"
         self.btn_back.disabled = True
-
         self.m.set_pause(False)
         self.is_job_started_already = True
-        log('Starting job...')
-
-        # Vac_fix. Not very tidy but will probably work.
-        # Also inject zUp-on-pause code if needed
```
Comment (LettyYeti) [inline src/asmcnc/skavaUI/screen_go.py:776]: .

File: src/asmcnc/skavaUI/screen_go.py [hunk 5]
```diff
@@ -539,425 +551,450 @@ def show_hide_yp_container(self, use_sc2):
...
+        log("Starting job...")
 
         def prep_gcode(job_to_modify):
             modified_gcode = []
-
-            # Spindle command??
```
Comment (LettyYeti) [inline src/asmcnc/skavaUI/screen_go.py:781]: .

Comment (LettyYeti) [reply src/asmcnc/skavaUI/screen_go.py:781]: and comments below also

File: src/asmcnc/skavaUI/screen_go.py [hunk 6]
```diff
@@ -539,425 +551,450 @@ def show_hide_yp_container(self, use_sc2):
...
-            if self.lift_z_on_job_pause and self.m.fw_can_operate_zUp_on_pause():  # extra 'and' as precaution
-                modified_gcode.append("M56 P0")  # append cleaned up gcode to object
-
+            if self.lift_z_on_job_pause and self.m.fw_can_operate_zUp_on_pause():
+                modified_gcode.append("M56 P0")
             return modified_gcode
 
-        # Check if job recovery is being performed
```
Comment (LettyYeti) [inline src/asmcnc/skavaUI/screen_go.py:804]: .

File: src/asmcnc/skavaUI/screen_go.py [hunk 7]
```diff
@@ -539,425 +551,450 @@ def show_hide_yp_container(self, use_sc2):
...
-            log('Job started ok from go screen...')
-
+            log("Job started ok from go screen...")
         except:
-            log('Job start from go screen failed!')
+            log("Job start from go screen failed!")
 
     def return_to_app(self):
-
-        if self.m.fw_can_operate_zUp_on_pause():  # precaution
```
Comment (LettyYeti) [inline src/asmcnc/skavaUI/screen_go.py:843]: .

File: src/asmcnc/skavaUI/screen_go.py [hunk 8]
```diff
@@ -539,425 +551,450 @@ def show_hide_yp_container(self, use_sc2):
...
-            self.m.send_any_gcode_command("M56 P0")  # disables Z lift on pause
+        if self.m.fw_can_operate_zUp_on_pause():
+            self.m.send_any_gcode_command("M56 P0")
         self.sm.current = self.cancel_to_screen
         self.m._grbl_soft_reset()
 
-    ### GLOBAL ENTER/LEAVE ACTIONS
```
Comment (LettyYeti) [inline src/asmcnc/skavaUI/screen_go.py:848]: .

File: src/asmcnc/skavaUI/screen_go.py [hunk 9]
```diff
@@ -539,425 +551,450 @@ def show_hide_yp_container(self, use_sc2):
...
+        if self.loop_for_feeds_and_speeds != None:
+            self.loop_for_feeds_and_speeds.cancel()
         if self.listen_for_pauses != None:
             self.listen_for_pauses.cancel()
             self.listen_for_pauses = None
-
         if self.yp_widget.yp_settings_popup:
             self.yp_widget.yp_settings_popup.dismiss()
 
-    ### SCREEN UPDATES
```
Comment (LettyYeti) [inline src/asmcnc/skavaUI/screen_go.py:860]: .

File: src/asmcnc/skavaUI/screen_go.py [hunk 10]
```diff
@@ -539,425 +551,450 @@ def show_hide_yp_container(self, use_sc2):
...
-                self.run_time_label.text = str(seconds) + " " + self.l.get_str("seconds")
+                self.run_time_label.text = (
+                    str(seconds) + " " + self.l.get_str("seconds")
+                )
         else:
             self.run_time_label.text = self.l.get_str("Waiting for job to be started")
 
     def poll_for_feeds_and_speeds(self, dt):
-
-        # Spindle speed and feed rate
```
Comment (LettyYeti) [inline src/asmcnc/skavaUI/screen_go.py:902]: .

File: src/asmcnc/skavaUI/screen_go.py [hunk 11]
```diff
@@ -539,425 +551,450 @@ def show_hide_yp_container(self, use_sc2):
...
-
-        if abs(self.feedOverride.feed_override_percentage - 100) > abs(self.feed_rate_max_percentage - 100):
+        if abs(self.feedOverride.feed_override_percentage - 100) > abs(
+            self.feed_rate_max_percentage - 100
+        ):
             self.feed_rate_max_percentage = self.feedOverride.feed_override_percentage
         if self.feedOverride.feed_absolute.text > self.feed_rate_max_absolute:
             self.feed_rate_max_absolute = self.feedOverride.feed_absolute.text
 
-    # Called from serial_connection if change in state seen
```
Comment (LettyYeti) [inline src/asmcnc/skavaUI/screen_go.py:918]: .

**Review by LettyYeti — CHANGES_REQUESTED** [review]
Mostly putting comments back, but also a few things on scaling & translations that have gone weird

File: src/asmcnc/core_UI/job_go/widgets/widget_disabled_yeti_pilot.py [hunk 12]
```diff
@@ -1,10 +1,13 @@
-from kivy.uix.widget import Widget
+from kivy.core.window import Window
 from kivy.lang import Builder
-from asmcnc.core_UI.job_go.screens.screen_spindle_health_check import SpindleHealthCheckActiveScreen
-from kivy.uix.scrollview import ScrollView
```
[the remark this answers is not in the document]
Comment (archiepeterjarvis) [reply src/asmcnc/core_UI/job_go/widgets/widget_disabled_yeti_pilot.py:4]: kivy builder doesn't require things to be imported in python

File: src/asmcnc/skavaUI/widget_feed_override.py [hunk 13]
```diff
@@ -1,181 +1,173 @@
...
+    BoxLayout:
+        size: self.parent.size
+        pos: self.parent.pos      
+
+        spacing:0.0*app.height
+        
+        orientation: "vertical"
+        
+        Button:
+            font_size: str(0.01875 * app.width) + 'sp'
```
[the remark this answers is not in the document]
Comment (archiepeterjarvis) [reply src/asmcnc/skavaUI/widget_feed_override.py:31]: I think this was the script doing it automatically

File: src/asmcnc/core_UI/job_go/popups/popup_yetipilot_settings.py [hunk 14]
```diff
@@ -97,368 +92,426 @@
...
+        advice_container_width = pop_width - dropdowns_container_width - (30.0/800.0*Window.width)
+        spindle_health_check_button_size = 150.0/800*Window.width
         spindle_health_check_button = BigSpindleHealthCheckButton()
-
-        transparent = [0,0,0,0]
-        subtle_white = [249 / 255., 249 / 255., 249 / 255., 1.]
-        blue = [33 / 255., 150 / 255., 243 / 255., 1.]
-        dark_grey = [51 / 255., 51 / 255., 51 / 255., 1.]
-
-        # Title
```
[the remark this answers is not in the document]
Comment (archiepeterjarvis) [reply src/asmcnc/core_UI/job_go/popups/popup_yetipilot_settings.py:156]: are these really needed? the line below is just title_string = ...

**Review by LettyYeti — APPROVED** [review]

**Pull request merged**